### ISSUE TITLE
NC 22: Circles migration too slow

### ISSUE THREAD

**Issue by svenseeberg** [nextcloud/circles#856]
We recently upgraded a Nextcloud with a 4 digit number of circles and 5 digit number of users to NC22. After the migration, as documentend in the known issues, the Circle memberships did not work. Just for clarity: that means that many shared files, Decks boards etc are not available for users.

First, we ran

`occ circles:maintenance --level 4`

which is recommended in the known issues section of the forum to facilitate a faster migration of Circles. In our case the command ran for about 3 days and in the end the circles were not migrated.

When looking into the source code, we found that the `circles:sync` command has some parameters that seemingly do what we want:
```
sudo -u www-data php -f /opt/nextcloud/occ circles:sync --migration [--force-run]
```
We're still waiting for the script to finish, but the migration seems to be making visible progress now.

However, the speed is again very limited. Currently, it migrates about 2 memberships per second. The total number of memberships is a 6 digit number.

To sum things up: The speed of the migration is very slow for large Nextcloud installations and the documentation for the problem is lacking. If Circles are widely used in an organization with a large NC installation, then this can be a blocking issue for a migration to NC22.

**Comment by svenseeberg** [comment]
We're quite confident that this is a caching problem as mentioned in https://help.nextcloud.com/t/circles-nc22-1-0-known-issues/121600. Does anyone now a quicker way to recreate the cache? The `circles:maintenance --level 4` command is extremely slow.

**Comment by svenseeberg** [comment]
It seems we are waiting for old circles to be transferred from the `oc_circle_circles` to the `oc_circles_circle`, which is probably done by the aforementioned command? Can someone confirm this?

**Comment by svenseeberg** [comment]
`occ circles:maintenance --level 4` is now finished and the "old" circles are not available. The answer to my previous question seems to be "no". It there any script that should facilitate a transition? If not, I would start programming something that migrates the existing circles into the new tables.

**Comment by ArtificialOwl** [comment]
Thanks for your report, I will see if there is a way to speed the migration process

**Comment by ArtificialOwl** [comment]
So, after a descent look on the migration process, I do not think of an easy/quick/clean way to improve it.
Unfortunatly, It will require more time that I can spend for an issue that occurs on really specific setup for a single-ran operation. 

But you are totally right, I should display this fact somewhere

**Comment by svenseeberg** [comment]
I think mentioning this in the changelog is enough to consider this issue "fixed". I expect that there are not that many Nextcooud installations with that many users and circles. Thanks for your work @ArtificialOwl !
